File: extract_pdf.js
const fs = require('fs');
const PDFParser = require('pdf2json');

const pdfParser = new PDFParser(this, 1);

pdfParser.on('pdfParser_dataError', errData => console.error(errData.parserError));
pdfParser.on('pdfParser_dataReady', pdfData => {
    fs.writeFileSync('c:\\Users\\lagsu\\OneDrive\\Escritorio\\Smartune\\Documentacion\\extracted_text.txt', pdfParser.getRawTextContent());
    console.log('Extraction completed');
});

pdfParser.loadPDF('c:\\Users\\lagsu\\OneDrive\\Escritorio\\Smartune\\Documentacion\\Documentacion Final Skeletune.pdf');

File: extract_pdf.py
import sys
try:
    from pypdf import PdfReader
except ImportError:
    import subprocess
    subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'pypdf'])
    from pypdf import PdfReader

reader = PdfReader(r'c:\Users\lagsu\OneDrive\Escritorio\Smartune\Documentacion\Documentacion Final Skeletune.pdf')
text = ''
for page in reader.pages:
    text += page.extract_text() + '\n'

with open(r'c:\Users\lagsu\OneDrive\Escritorio\Smartune\Documentacion\extracted_text.txt', 'w', encoding='utf-8') as f:
    f.write(text)
print('Extraction completed')
